lib/contract/deploy_asset_contract.ex

```diff
@@ -25,15 +25,20 @@ defmodule Soroban.Contract.DeployAssetContract do
   @type invoke_host_function :: InvokeHostFunction.t()
   @type secret_key :: binary()
   @type send_response :: {:ok, SendTransactionResponse.t()}
+  @type issuer_pub_key :: binary() | nil
 
-  @spec deploy(asset_code :: asset_code(), secret_key :: secret_key()) :: send_response()
-  def deploy(asset_code, secret_key) do
+  @spec deploy(
+          asset_code :: asset_code(),
+          secret_key :: secret_key(),
+          issuer_pub_key :: issuer_pub_key()
+        ) :: send_response()
+  def deploy(asset_code, secret_key, issuer_pub_key \\ nil) do
```

Comment:
@MarioRodriguezS some suggestions here:

1. IMO we should reorder:
- `asset_code`
- `asset_issuer`
- `secret_key`

2. Notice `issuer_pub_key` can be renamed to `asset_issuer`.

3. I think we should not allow `asset_issuer` to be optional. The `asset_issuer` should always be specified for clarity. An asset is always made up of both asset code and issuer.
  Inferring the issuer from the secret can be confusing.